### PR TITLE
Add recipe for `mirror` and make it required for `self`

### PR DESCRIPTION
**Pull request by byorgey**
The `whoami` command used to be more interesting back when robot names were used to uniquely identify them.  It seemed fitting to make `mirror` required for `self`, which now plays the same role.  I also added a recipe for `mirror` which we lacked (`glass` + `silver`) and made a new type of `deep mine` where we can mine `gold`, `silver`, and `mithril`.

### CODE REVIEW

**Comment by byorgey** [comment]
Oh, good idea, I like it!  We can have mithril drop very occasionally from a deep mine. =)  I will change it.

**Comment by byorgey** [comment]
I didn't add lava but we could definitely add that later.

File: src/Swarm/Language/Capability.hs [hunk 1]
```diff
@@ -304,7 +305,6 @@ constCaps =
     Appear -> [CAppear] -- paint?
     Whereami -> [CSenseloc] -- GPS?
     Random -> [CRandom] -- randomness device (with bitcoins)?
-    Whoami -> [CWhoami] -- mirror, needs a recipe
```
Comment (xsebek) [inline src/Swarm/Language/Capability.hs:307]: I think we might have had a similar idea, so I caused you a merge conflict. Sorry about that :sweat_smile:

**Review by xsebek — APPROVED** [review]
Amazing, the entity descriptions are perfect. :mage_man:

**Comment by byorgey** [comment]
I was thinking that in some kind of difficult mode we add later, drilling in a deep mine can very occasionally cause a balrog to spawn, who then wanders around setting everything on fire. 😀 🔥

File: src/Swarm/Language/Capability.hs (hunk 1 above)
Comment (byorgey) [reply src/Swarm/Language/Capability.hs:307]: Hah, no worries.

**Pull request merged**